genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_proxy.go

```diff
@@ -0,0 +1,172 @@
+type getEdgeGroupByNameFunc func(ctx context.Context, p *edgeGroupProxy, edgeGroupName string, managed bool) (*[]platformclientv2.Edgegroup, error)
+
+type edgeGroupProxy struct {
+	clientConfig *platformclientv2.Configuration
+	edgesApi     *platformclientv2.TelephonyProvidersEdgeApi
+
+	getEdgeGroupByIdAttr   getEdgeGroupByIdFunc
+	deleteEdgeGroupAttr    deleteEdgeGroupFunc
+	updateEdgeGroupAttr    updateEdgeGroupFunc
+	createEdgeGroupAttr    createEdgeGroupFunc
+	getAllEdgeGroupsAttr   getAllEdgeGroupsFunc
+	getEdgeGroupByNameAttr getEdgeGroupByNameFunc
+}
+
+func newEdgeGroupProxy(clientConfig *platformclientv2.Configuration) *edgeGroupProxy {
+	edgesApi := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(clientConfig)
+
+	return &edgeGroupProxy{
+		clientConfig: clientConfig,
+		edgesApi:     edgesApi,
+
+		getEdgeGroupByIdAttr:   getEdgeGroupByIdFn,
+		deleteEdgeGroupAttr:    deleteEdgeGroupFn,
+		updateEdgeGroupAttr:    updateEdgeGroupFn,
+		createEdgeGroupAttr:    createEdgeGroupFn,
+		getAllEdgeGroupsAttr:   getAllEdgeGroupsFn,
+		getEdgeGroupByNameAttr: getEdgeGroupByNameFn,
+	}
+}
+
+func getEdgeGroupProxy(clientConfig *platformclientv2.Configuration) *edgeGroupProxy {
+	if internalProxy == nil {
+		internalProxy = newEdgeGroupProxy(clientConfig)
+	}
+	return internalProxy
+}
+
+func (p *edgeGroupProxy) getEdgeGroupById(ctx context.Context, edgeGroupId string) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	return p.getEdgeGroupByIdAttr(ctx, p, edgeGroupId)
+}
+
+func (p *edgeGroupProxy) deleteEdgeGroup(ctx context.Context, edgeGroupId string) (*platformclientv2.APIResponse, error) {
+	return p.deleteEdgeGroupAttr(ctx, p, edgeGroupId)
+}
+
+func (p *edgeGroupProxy) updateEdgeGroup(ctx context.Context, edgeGroupId string, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	return p.updateEdgeGroupAttr(ctx, p, edgeGroupId, body)
+}
+
+func (p *edgeGroupProxy) createEdgeGroup(ctx context.Context, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	return p.createEdgeGroupAttr(ctx, p, body)
+}
+
+func (p *edgeGroupProxy) getAllEdgeGroups(ctx context.Context) (*[]platformclientv2.Edgegroup, error) {
+	return p.getAllEdgeGroupsAttr(ctx, p)
+}
+
+func (p *edgeGroupProxy) getEdgeGroupByName(ctx context.Context, edgeGroupName string, managed bool) (*[]platformclientv2.Edgegroup, error) {
+	return p.getEdgeGroupByNameAttr(ctx, p, edgeGroupName, managed)
+}
+
+func getEdgeGroupByNameFn(ctx context.Context, p *edgeGroupProxy, edgeGroupName string, managed bool) (*[]platformclientv2.Edgegroup, error) {
+	const pageSize = 100
+	var pageNum = 1
+	var allEdgeGroups []platformclientv2.Edgegroup
+
+	edgeGroups, _, err := p.edgesApi.GetTelephonyProvidersEdgesEdgegroups(pageSize, pageNum, edgeGroupName, "", managed)
```

Comment:
You don't need to repeat this looping logic, calling `getAllEdgeGroupsFn` will return all the egeGroups

Reply:
actually, we're passing the edgegroupName and managed parameter here. so this is slightly different